### PR TITLE
Problem: Missing detection of standards greater than C++11

### PR DESCRIPTION
**Pull request by gummif**
Solution: Add macros detecting C++14 and C++17

### CODE REVIEW

File: zmq.hpp [hunk 1]
```diff
@@ -26,22 +26,37 @@
...
 #elif defined(_MSC_VER)
 #define ZMQ_DEPRECATED(msg) __declspec(deprecated(msg))
 #elif defined(__GNUC__)
 #define ZMQ_DEPRECATED(msg) __attribute__((deprecated(msg)))
 #endif
 
-#if (__cplusplus >= 201103L) || (defined(_MSC_VER) && (_MSC_VER >= 1900))
-#define ZMQ_CPP11
+#if defined(ZMQ_CPP17)
+#define ZMQ_NODISCARD [[nodiscard]]
```
Comment (sigiesec) [inline zmq.hpp]: This adds a definition of ZMQ_NODISCARD, but it this macro is not used anywhere. Are you working on a adding uses of this to zmq.hpp?

Comment (gummif) [reply zmq.hpp]: Yes, `bool empty()` for message_t would be useful, where nodiscard would be appropriate.

**Comment by sigiesec** [comment]
Sure. I think there are many other methods as well where nodiscard would make sense. 

My question was if you intend to add these attributes in another PR soon, so we don't have a state with this unused definition around for long.

File: zmq.hpp (hunk 1 above)
Comment (sigiesec) [reply zmq.hpp]: Sure. I think there are many other methods as well where nodiscard would make sense.

My question was if you intend to add these attributes in another PR soon, so we don't have a state with this unused definition around for long.

Comment (gummif) [reply zmq.hpp]: Yes, I could do that within a couple of days.

Comment (sigiesec) [reply zmq.hpp]: Great, I will merge this PR now.

**Pull request merged**